doc/src/tutorial/codegen_explainatory.rst

```diff
@@ -0,0 +1,458 @@
+==========================
+Code Generation from Sympy
+==========================
+Several submodules in Sympy are provided to generate directly compilable 
+code from Sympy expressions. Furthermore, there are submodules provided 
+that help use other numeric systems as backend for construction and 
+manipulation of mathematical expressions to speed up the process.
+
+Autowrap
+--------
+Autowrap automatically generates code, writes it to disk, compiles it, 
+and imports it into the current session. Main functions of this module are 
+``autowrap``, ``binary_function``, and ``ufuncify``.
+
+It also automatically converts expressions containing ``Indexed`` objects 
+into summations. The classes IndexedBase, Indexed and Idx represent a matrix 
+element M[i, j]. See :ref:`tensor_module` for more on this.
+``autowrap`` creates a wrapper using f2py or Cython and creates a numerical 
+function.
+
+.. note:: The ``autowrap`` callable is not in the sympy namespace automatically,
+   to use it you must first import ``autowrap`` from ``sympy.utilities.autowrap``
+
+
+The callable returned from autowrap() is a binary python function, not a 
+SymPy object. For example::
+
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.utilities.autowrap import autowrap
+    >>> expr = ((x - y + z)**(13)).expand()
+    >>> binary_func = autowrap(expr)
+    >>> binary_func(1, 4, 2)
+    -1.0
+
+The various flags available with autowrap() help to modify the services 
+provided by the method. 
+The argument ‘tempdir’ tells autowrap to compile the code in a specific 
+directory, and leave the files intact when finished. For instance::
+
+    >>> from sympy.utilities.autowrap import autowrap
+    >>> from sympy.physics.qho_1d import psi_n
+    >>> from sympy import IndexedBase, Idx
+    >>> from sympy import Eq
+    >>> from sympy import symbols
+    >>> x=IndexedBase('x')
+    >>> y=IndexedBase('y')
+    >>> m=symbols('m', integer=True)
+    >>> i = Idx('i', m)
+    >>> a,omega= symbols('a, omega')
+    >>> qho=autowrap(Eq(y[i], psi_n(0, x[i], m, omega)), tempdir='/home/thd/Documents/tmp')
```

Comment:
The doctests will run this on Travis. You should either use a directory in the current working directory or one in `/tmp/`. This may not work as is and it is better to make it more general.